File: 95/logger.nim
import times, strformat, locks

var
  logLock: Lock

proc initLogger*() =
  initLock(logLock)

proc logMessage*(level: LogLevel, message: string) =
  let timestamp = getTime().format("yyyy-MM-dd HH:mm:ss")
  let levelStr = case level
    of DEBUG: "DEBUG"
    of INFO: "INFO"
    of WARN: "WARN"
    of ERROR: "ERROR"
  
  let logLine = fmt"[{timestamp}] [{levelStr}] {message}"
  
  withLock logLock:
    echo logLine
    flushFile(stdout)

proc debug*(message: string) = logMessage(DEBUG, message)
proc info*(message: string) = logMessage(INFO, message)
proc warn*(message: string) = logMessage(WARN, message)
proc error*(message: string) = logMessage(ERROR, message)

File: 426/pkg/utils/logger.go
package utils

import (
	"log"
	"os"
)

type Logger struct {
	debug   *log.Logger
	info    *log.Logger
	warning *log.Logger
	error   *log.Logger
	level   string
}

func NewLogger(level string) *Logger {
	return &Logger{
		debug:   log.New(os.Stdout, "[DEBUG] ", log.Ldate|log.Ltime|log.Lshortfile),
		info:    log.New(os.Stdout, "[INFO] ", log.Ldate|log.Ltime),
		warning: log.New(os.Stdout, "[WARN] ", log.Ldate|log.Ltime),
		error:   log.New(os.Stderr, "[ERROR] ", log.Ldate|log.Ltime|log.Lshortfile),
		level:   level,
	}
}

func (l *Logger) Debug(format string, v ...interface{}) {
	if l.level == "debug" {
		l.debug.Printf(format, v...)
	}
}

func (l *Logger) Info(format string, v ...interface{}) {
	if l.level == "debug" || l.level == "info" {
		l.info.Printf(format, v...)
	}
}

func (l *Logger) Warning(format string, v ...interface{}) {
	if l.level != "error" {
		l.warning.Printf(format, v...)
	}
}

func (l *Logger) Error(format string, v ...interface{}) {
	l.error.Printf(format, v...)
}

func (l *Logger) Fatal(format string, v ...interface{}) {
	l.error.Fatalf(format, v...)
}
